backend/src/services/billit/tests/api-client.test.ts

```diff
@@ -0,0 +1,325 @@
+    });
+
+    it('should throw error when API returns non-ok response', async () => {
+      const errorMessage: string = 'Invalid order data';
+      mockFetch.mockResolvedValueOnce({
+        ok: false,
+        status: 400,
+        text: () => Promise.resolve(errorMessage),
+      } as any);
+
+      await expect(apiClient.createOrder(createOrderRequest, '2024-001')).rejects.toThrow(
+        `Failed to create order at Billit: ${errorMessage}`,
+      );
+
+      expect(logger.error).toHaveBeenCalledWith(`Billit createOrder failed: 400 - ${errorMessage}`);
+    });
+
+    it('should include correct idempotency key', async () => {
+      mockFetch.mockResolvedValueOnce({
+        ok: true,
+        text: () => Promise.resolve('123456'),
+      } as any);
+
+      await apiClient.createOrder(createOrderRequest, 'unique-key-123');
+
+      expect(mockFetch).toHaveBeenCalledWith(
+        `${apiConfig.apiUrl}/orders`,
+        expect.objectContaining({headers: expect.objectContaining({'Idempotency-Key': 'unique-key-123'})}),
+      );
+    });
+  });
+
+  describe('sendInvoice', () => {
+    it('should send invoice via Peppol successfully', async () => {
+      const request: SendInvoiceRequest = {
+        TransportType: 'Peppol',
+        OrderIDs: [123456, 123457],
+      };
+
+      mockFetch.mockResolvedValueOnce({ok: true} as any);
+
+      await apiClient.sendInvoice(request, 'send-123');
+
+      expect(mockFetch).toHaveBeenCalledWith(
+        `${apiConfig.apiUrl}/orders/commands/send`,
+        expect.objectContaining({
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            ApiKey: apiConfig.apiKey,
+            PartyID: apiConfig.partyId,
+            ContextPartyID: apiConfig.contextPartyId,
+            'Idempotency-Key': 'send-123',
+          },
+          body: JSON.stringify(request),
+        }),
+      );
+      expect(logger.info).toHaveBeenCalledWith(`Invoice(s) ${request.OrderIDs.join(', ')} sent via ${request.TransportType}`);
+    });
+
+    it('should send invoice via SMTP successfully', async () => {
+      const request: SendInvoiceRequest = {
+        TransportType: 'SMTP',
+        OrderIDs: [123456],
+      };
+
+      mockFetch.mockResolvedValueOnce({ok: true} as any);
+
+      await apiClient.sendInvoice(request, 'send-124');
+
+      expect(mockFetch).toHaveBeenCalledWith(
+        `${apiConfig.apiUrl}/orders/commands/send`,
+        expect.objectContaining({
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            ApiKey: apiConfig.apiKey,
+            PartyID: apiConfig.partyId,
+            ContextPartyID: apiConfig.contextPartyId,
+            'Idempotency-Key': 'send-124',
+          },
+          body: JSON.stringify(request),
+        }),
+      );
+      expect(logger.info).toHaveBeenCalledWith(`Invoice(s) ${request.OrderIDs.join(', ')} sent via SMTP`);
+    });
```

Comment:
Watch out for tautological tests

Check bijvoorbeeld: https://randycoulman.com/blog/2016/12/20/tautological-tests/

Reply:
Ik heb het hieronder ergens gezet. Het heeft geen zin om testen voor API calls te schrijven, je zou dit kunnen testen door testen te schrijven voor de controller functie.

Wat hier wel interessant zou kunnen zijn, zou zijn om echte reponses die de API effectief terug geeft te testen (testen als documentatie). Want ik vermoed, zeker omdat de testen door AI gemaakt zijn, dat de error-cases verzonnen zijn door AI, en dat de BillitAPI waarschijnlijk niet zo'n responses teruggeeft. De testen hebben dus geen waarde.